projects/lookup/src/modules/lookup/lookup.component.ts

```diff
@@ -529,6 +513,63 @@ export class SkyLookupComponent
     }
   }
 
+  private createNativePickerInstance(initialSearch: string): SkyModalInstance {
+    let initialValue = this.value;
+    const modalConfig = this.showMoreConfig?.nativePickerConfig || {};
+
+    if (!modalConfig.itemTemplate) {
+      modalConfig.itemTemplate = this.searchResultTemplate;
+    }
+
+    let contextProviderType:
+      | typeof SkyLookupShowMoreNativePickerContext
+      | typeof SkyLookupShowMoreNativePickerAsyncContext;
+
+    let contextProviderValue:
+      | SkyLookupShowMoreNativePickerContext
+      | SkyLookupShowMoreNativePickerAsyncContext;
+
+    let modalComponentType:
+      | typeof SkyLookupShowMoreModalComponent
+      | typeof SkyLookupShowMoreAsyncModalComponent;
+
+    if (this.searchAsync.observers.length > 0) {
+      contextProviderType = SkyLookupShowMoreNativePickerAsyncContext;
+      modalComponentType = SkyLookupShowMoreAsyncModalComponent;
+
+      contextProviderValue = new SkyLookupShowMoreNativePickerAsyncContext();
+      contextProviderValue.idProperty = this.idProperty;
+      contextProviderValue.searchAsync = (args) => {
+        this.searchAsync.emit(args);
+        return args.result;
+      };
+    } else {
+      contextProviderType = SkyLookupShowMoreNativePickerContext;
+      modalComponentType = SkyLookupShowMoreModalComponent;
+
+      contextProviderValue = new SkyLookupShowMoreNativePickerContext();
+      contextProviderValue.items = this.data;
+      contextProviderValue.search = this.search;
+    }
+
+    contextProviderValue.descriptorProperty = this.descriptorProperty;
+    contextProviderValue.initialSearch = initialSearch;
+    contextProviderValue.initialValue = initialValue;
```

Comment:
That would only protect top-level properties from being altered on the original value, but not deeper properties, e.g. `foo.bar = 1` would not change properties on the original object, but `foo.bar.baz = 1` would still alter the original object. Since this is all internal, I think it's safe to keep it as-is for now.